core/modules/replica/Controller.h

```diff
@@ -0,0 +1,687 @@
+     *
+     * @param workerName     - the name of a worker node where the replica will be deleted
+     * @param database       - database name
+     * @param chunk          - the chunk number
+     * @param onFinish       - an optional callback function to be called upon the completion of the request
+     * @param priority       - a priority level of the request
+     * @param keepTracking   - keep tracking the request before it finishes or fails
+     * @param allowDuplicate - follow a previously made request if the current one duplicates it
+     * @param jobId          - an optional identifier of a job issed the request
+     * @param requestExpirationIvalSec
+     *                       - an optional parameter (if differs from 0)
+     *                         allowing to override the default value of the corresponding
+     *                         parameter from the Configuration.
+     *
+     * @return a pointer to the replication request
+     */
+    DeleteRequest_pointer deleteReplica(std::string const& workerName,
+                                        std::string const& database,
+                                        unsigned int chunk,
+                                        DeleteRequest_callback_type onFinish=nullptr,
+                                        int  priority=0,
+                                        bool keepTracking=true,
+                                        bool allowDuplicate=true,
+                                        std::string const& jobId="",
+                                        unsigned int requestExpirationIvalSec=0);
+
+    /**
+     * Initiate a new replica lookup request.
+     *
+     * PERFORMANCE NOTE: enabling 'computeCheckSum' will require reading each file.
+     *                   Hence this will slow down the operation, and it may also
+     *                   affcet the overall perfromance of Qserv on the corresponding
+     *                   worker node.
+     *
+     * @param workerName      - the name of a worker node where the replica is located
+     * @param database        - database name
+     * @param chunk           - the chunk number
+     * @param onFinish        - an optional callback function to be called upon the completion of the request
+     * @param priority        - a priority level of the request
+     * @param computeCheckSum - tell a worker server to compute check/control sum on each file
+     * @param keepTracking    - keep tracking the request before it finishes or fails
+     * @param jobId           - an optional identifier of a job issed the request
+     * @param requestExpirationIvalSec
+     *                         - an optional parameter (if differs from 0)
+     *                           allowing to override the default value of the corresponding
+     *                           parameter from the Configuration.
+     *
+     * @return a pointer to the replication request
+     */
```

Comment:
so much of this documentation is copied & pasted - it seems like it might be more efficient for the reader to have the parameter names in the class documentation, and leave the parameters off the function documentation?

Reply:
This (letting each method to have its own self-container set of documentation) doesn't bother me as much as potential "bookkeeping" problems when/if moving the documentation into class header. Ot's much easier to ensure the consistency of a method signature vs its documentation when they stay closer to act other.